MediaBrowser.Controller/Resolvers/Scanner.cs

```diff
@@ -0,0 +1,38 @@
+#pragma warning disable CS1591
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Jellyfin.Data.Enums;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Controller.Providers;
+using MediaBrowser.Model.IO;
+
+namespace MediaBrowser.Controller.Resolvers
+{
+    public abstract class Scanner : IScanner
+    {
+        public bool Enabled { get; set; } = false;
+
+        public bool Default { get; set; } = false;
+
+        public ResolverPriority Priority => ResolverPriority.Plugin;
+
+        protected BaseItem ApplyMetadata(BaseItem t)
+        {
+            throw new NotImplementedException();
```

Comment:
This would prevent safe use of calling `t = base.ApplyMetadata(t);` in multiple-layer derived resolver classes. Do we explicitly want to discourage that sort of coding?

Perhaps better to just make it this 
```c#
return t;
```

Reply:
yes.

this is the base function/only function that actually does something here. There is no other reason to implement this base class, so why not just make this function abstract itself?

```suggestion
            protected abstract BaseItem ApplyMetadata(BaseItem t);
```